src/base/projects/View.svelte

```diff
@@ -37,99 +88,83 @@
-      <Route path="/history/*" let:params let:location>
-        <ProjectRoute
-          route={params["*"]}
-          content={ProjectContent.History}
-          {location}
-          {peer}
-          {project}
-          {wallet} />
-      </Route>
-
-      <Route path="/commits/:commit" let:params>
-        <ProjectRoute
-          revision={params.commit}
-          content={ProjectContent.Commit}
-          {peer}
-          {project}
-          {wallet} />
-      </Route>
-      <Route path="/commits/*" let:params let:location>
-        <ProjectRoute
-          route={params["*"]}
-          content={ProjectContent.Commit}
-          {location}
-          {peer}
-          {project}
-          {wallet} />
-      </Route>
-
-      <Route path="/issues" let:location>
-        <ProjectRoute
-          content={ProjectContent.Issues}
-          {peer}
-          {project}
-          {location}
-          {wallet} />
-      </Route>
-      <Route path="/issues/:issue" let:params let:location>
-        <ProjectRoute
-          content={ProjectContent.Issue}
-          issue={params.issue}
-          {peer}
-          {project}
-          {location}
-          {wallet} />
-      </Route>
+    {#await project.getRoot(revision)}
+      <Loading center />
+    {:then { tree, commit }}
+      <Header {tree} {commit} {project} />
 
-      <Route path="/patches">
-        <ProjectRoute
-          content={ProjectContent.Patches}
-          {peer}
-          {project}
-          {wallet} />
-      </Route>
-      <Route path="/patches/:patch" let:params>
-        <ProjectRoute
-          content={ProjectContent.Patch}
-          patch={params.patch}
-          {peer}
-          {project}
-          {wallet} />
-      </Route>
-    </Router>
+      {#if content === "tree"}
+        <Browser {project} {commit} {tree} />
+      {:else if content === "commits"}
+        {#await proj.Project.getCommits( project.urn, project.seed.api, { parent: commit, verified: true }, )}
+          <Loading center />
+        {:then history}
+          <History {project} {history} />
+        {:catch e}
+          <div class="message">
+            <Message error>{e.message}</Message>
+          </div>
+        {/await}
+      {:else if content === "commit"}
+        {#await project.getCommit(commit)}
+          <Loading center />
+        {:then commit}
+          <Commit {project} {commit} />
+        {:catch e}
+          <div class="message">
+            <Message error>{e.message}</Message>
+          </div>
+        {/await}
+      {:else if content === "issues"}
+        {#await issue.Issue.getIssues(project.urn, project.seed.api)}
+          <Loading center />
+        {:then issues}
+          <Issues {project} state={issueFilter} {wallet} {issues} />
+        {:catch e}
+          <div class="message">
+            <Message error>{e.message}</Message>
+          </div>
+        {/await}
+      {:else if content === "issue" && $activeRoute.params.issue}
```

Comment:
Why are there singlular versions of certain resources now?

Reply:
We started using singular versions of certain resources, to add type safety to the different route types and limit certain route params to `issues` and others to the `issue` route.
Else we would have the `issue` param in one case as `null` and in another as a `string`

Reply:
Ah, so this doesn't affect the URL?

Reply:
Currently it does affect the URL since we match the resource name used in the `Route` type with the URL.
But I think we could to maintain the same resource between a listing and a detail view by tweaking the parsing functions a bit

Reply:
Ok pushed be338c674db2bdc44303ad223c2d47de147e117e which reverts the project resources back to the current state, with minimal changes to the `pathToRoute` and `routeToPath` functions